share/github-backup-utils/ghe-restore-es-audit-log

```diff
@@ -43,22 +43,31 @@ fi
 # Only restore indices that don't exist and the last two months' indices.
 for index in $indices; do
   if ! ghe-ssh "$GHE_HOSTNAME" "curl -f -s -XGET http://localhost:9201/$index > /dev/null" || [[ $index =~ $last_month ]] || [[ $index =~ $current_month ]]; then
-    if $CLUSTER || [ -n "$configured" ]; then
-      ghe_verbose "* Restoring $index"
-      gzip -dc $GHE_DATA_DIR/$GHE_RESTORE_SNAPSHOT/audit-log/$index.gz | ghe-ssh "$GHE_HOSTNAME" "/usr/local/share/enterprise/ghe-es-load-json 'http://localhost:9201/$index'" 1>&3
-    else
-      echo "$index.gz" >> $tmp_list
-    fi
+    echo "$index.gz" >> $tmp_list
   fi
 done
 
 if [ -s "$tmp_list" ]; then
+  ghe-ssh "$GHE_HOSTNAME" -- "sudo mkdir -p '$GHE_REMOTE_DATA_USER_DIR/elasticsearch-restore'" 1>&3
+  ghe-ssh "$GHE_HOSTNAME" -- "sudo chown elasticsearch:elasticsearch '$GHE_REMOTE_DATA_USER_DIR/elasticsearch-restore'" 1>&3
+
   ghe-rsync -avz --delete \
     -e "ghe-ssh -p $(ssh_port_part "$GHE_HOSTNAME")" \
     --rsync-path="sudo -u elasticsearch rsync" \
     --files-from=$tmp_list \
     "$GHE_DATA_DIR/$GHE_RESTORE_SNAPSHOT/audit-log/" \
     "$(ssh_host_part "$GHE_HOSTNAME"):$GHE_REMOTE_DATA_USER_DIR/elasticsearch-restore/" 1>&3
+
+  if $CLUSTER || [ -n "$configured" ]; then
+    for index in $(cat $tmp_list | sed 's/\.gz$//g'); do
+      ghe_verbose "* Restoring $index"
+      echo "export PATH=\$PATH:/usr/local/share/enterprise && gzip -dc $GHE_REMOTE_DATA_USER_DIR/elasticsearch-restore/$index | ghe-es-load-json 'http://localhost:9201/$index'" |
+      ghe-ssh "$GHE_HOSTNAME" -- /bin/bash 1>&3
+    done
+  fi
+
+  ghe-ssh "$GHE_HOSTNAME" -- "sudo rm $GHE_REMOTE_DATA_USER_DIR/elasticsearch-restore/*.gz" 1>&3
```

Comment:
Doesn't this `rm` need to happen as part of the `if $CLUSTER || [ -n "$configured" ];` block?

Reply:
Yes and no. Yes cos that's about the only time it'll need to be run. No cos with a `-f` it'll work in all cases 😄

Reply:
Actually, ignore me. You're right. Deleting it here would mean the logs are removed for unconfigured instances before they can be imported. Changing.